docroot/modules/reusables/invisible_recaptcha/src/InvisibleRecaptcha.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\invisible_recaptcha;
+
+use Drupal\Core\State\StateInterface;
+
+/**
+ * Manages the invisible recaptcha actions.
+ */
+class InvisibleRecaptcha {
+
+  /**
+   * The api address for checking the invisible recaptcha key.
+   *
+   * @var string
+   */
+  const RECAPTCHA_API = 'https://www.google.com/recaptcha/api/siteverify?';
+
+  /**
+   * The state service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * The site key for client side integration.
+   *
+   * @var string
+   */
+  private $siteKey;
+
+  /**
+   * The secret key for server side integration.
+   *
+   * @var string
+   */
+  private $secretKey;
+
+  /**
+   * InvisibleRecaptcha constructor.
+   *
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The state service.
+   */
+  public function __construct(StateInterface $state) {
+    $this->state = $state;
+
+    $site_key = $this->state->get('invisible_recaptcha.site_key');
```

Comment:
I think that it should be better to set the secret key and site key as an argument for validation method or to be set with setters because as it now violates the open/closed principle.